File: sequence/sequence.go
package sequence

import (
	"math"
	"math/big"
)

func Sequence(index *big.Int) big.Int {
	index.Quo(index, big.NewInt(2))
	index.Add(index, big.NewInt(1))

	return *index
}

func ModSequence(index *big.Int) (big.Int, int64) {
	indexClone := new(big.Int).Set(index)
	var mod big.Int
	mod.Mod(indexClone, big.NewInt(4))
	
	modVal := mod.Int64()
	
	switch modVal {
	case 0:
		indexClone.Add(indexClone, big.NewInt(2))
	case 1:
		indexClone.Add(indexClone, big.NewInt(3))
	case 2:
		indexClone.Add(indexClone, big.NewInt(4))
	case 3:
		indexClone.Add(indexClone, big.NewInt(5))
	}
	indexClone.Div(indexClone, big.NewInt(2))
	
	return *indexClone, modVal
}

func FloatSequence(index float64) (float64, int64) {
	index = index / 2
	index = index + 1
	index = math.Round(index)
	return index, int64(index)
}

func BigFloatSequence(index *big.Float) *big.Float {
	index.Quo(index, big.NewFloat(2))
	index.Add(index, big.NewFloat(1))
	indexFloat, _ := index.Float64()
	return big.NewFloat(math.Round(indexFloat))
}

func main() {
	FloatSequence(545421)
}

File: sequence/sequence.js
const __ = require("underscore");

// [1, 2, 3, 4, 3, 4, 5, 6, 5, 6, 7, 8]
function sequence_recursive(limit = 10, prev = 0, next = 1, values = []) {
    if ((values.length - 1) === limit) {
        return values;
    }
    
    if (next < 4) {
        next = 1;
    }

    values.push(prev + next);
    return sequence_recursive(limit, prev + 2, next + 1, values);
}

// console.log(sequence_recursive(76));

function sequence_efficient(index = 10) {
    return Math.round(index - (index / 2)) + 1;
}
// 87123641123172368
// 81239812739128371

function sequence(limit = 10) {
    let index = 0;
    for (let i = 81239812739128360; true ; i += 2) {
        for (let j = 1; j <= 4; j++) {
            if (index === limit)
                return (i + j);
            index++;
        }
    }
}

function sequence_inneficient(limit = 10) {
    const values = []
    for (let i = 0; true ; i += 2) {
        for (let j = 1; j <= 4; j++) {
            values.push({ i, j, value: i + j });
            if ((values.length - 1) === limit) {
                return values;
            }
        }
    }
}

console.log(__.chunk(sequence_inneficient(50), 4));

module.exports = sequence_efficient;